inlong-manager/manager-schedule/src/main/java/org/apache/inlong/manager/schedule/dolphinscheduler/DolphinScheduleUtils.java

```diff
@@ -0,0 +1,719 @@
+import static org.apache.inlong.manager.schedule.dolphinscheduler.DolphinScheduleConstants.DS_SCHEDULE_DEF;
+import static org.apache.inlong.manager.schedule.dolphinscheduler.DolphinScheduleConstants.DS_SEARCH_VAL;
+import static org.apache.inlong.manager.schedule.dolphinscheduler.DolphinScheduleConstants.DS_TASK_DEFINITION;
+import static org.apache.inlong.manager.schedule.dolphinscheduler.DolphinScheduleConstants.DS_TASK_GEN_NUM;
+import static org.apache.inlong.manager.schedule.dolphinscheduler.DolphinScheduleConstants.DS_TASK_RELATION;
+import static org.apache.inlong.manager.schedule.dolphinscheduler.DolphinScheduleConstants.DS_TOKEN;
+
+/**
+ * A utility class for interacting with DolphinScheduler API. This class includes methods
+ * for creating, updating, and deleting projects, tasks, and process definitions in DolphinScheduler.
+ * It also provides methods for generating schedules, calculating offsets, and executing HTTP requests.
+ * This class leverages the OkHttp library for HTTP interactions and Gson for JSON parsing and serialization.
+ */
+public class DolphinScheduleUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DolphinScheduleEngine.class);
+
+    private static final long MILLIS_IN_SECOND = 1000L;
+    private static final long MILLIS_IN_MINUTE = 60 * MILLIS_IN_SECOND;
+    private static final long MILLIS_IN_HOUR = 60 * MILLIS_IN_MINUTE;
+    private static final long MILLIS_IN_DAY = 24 * MILLIS_IN_HOUR;
+    private static final long MILLIS_IN_WEEK = 7 * MILLIS_IN_DAY;
+    private static final long MILLIS_IN_MONTH = 30 * MILLIS_IN_DAY;
+    private static final long MILLIS_IN_YEAR = 365 * MILLIS_IN_DAY;
+    private static final String HTTP_POST = "POST";
+    private static final String HTTP_GET = "GET";
+    private static final String HTTP_PUT = "PUT";
+    private static final String HTTP_DELETE = "DELETE";
+    private static final String CONTENT_TYPE = "Content-Type: application/json; charset=utf-8";
+    private static final String SHELL_REQUEST_API = "/inlong/manager/api/group/submitOfflineJob";
+    private final OkHttpClient client;
+    private final ObjectMapper objectMapper;
+
+    /**
+     * Constructs a new instance of the DolphinScheduleUtils class.
+     * Initializes the OkHttpClient and Gson instances.
+     */
+    public DolphinScheduleUtils() {
+        this.client = new OkHttpClient();
+        this.objectMapper = new ObjectMapper();
+    }
+
+    /**
+     * Checks the uniqueness of a project ID based on the given search value.
+     *
+     * @param url       The base URL of the DolphinScheduler API.
+     * @param token     The authentication token to be used in the request header.
+     * @param searchVal The name of the project to search for.
+     * @return The unique project ID if found, or 0 if not found or an error occurs.
+     */
+    public long checkAndGetUniqueId(String url, String token, String searchVal) {
```

Comment:
As a `Utils` class, it should not be used through the new method. It is recommended to refer to HttpUtils.

Reply:
done